### DISCUSSION TITLE
Use 9.0 or 8.0 language features in project, targets to .NET Framework 4.7.2

### DISCUSSION THREAD

**Discussion by ViIvanov** [dotnet/roslyn#50378] [Q&A] [answered]
Hello,

I want to use some new language features in a project, that targets to .NET Framework 4.7.2.
I understand, that some features, like default interface methods and covariant return types will not work. I want to use only features, not related to runtime, like pattern matching, local and static local functions, etc.

It works fine, but is it supported by Microsoft? Could not find any "official" page related to this.

**Comment by PathogenDavid** [comment, marked as answer]
> It works fine, but is it supported by Microsoft?

No, it is not officially supported. From [C# language versioning](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version):

> C# 9.0 is supported only on .NET 5 and newer versions.

-----------

If you don't care about official support, [here's a list of what works and what doesn't in C# 9.0](https://github.com/dotnet/roslyn/discussions/47701#discussioncomment-79566). (This list does not include C# 8.0 features, which are also not officially supported on 4.7.2.)

**Reply by huoyaoyuan** [reply]
I think there should be more clarification for the term "support". I think it as "take effort to make everything working". But features without additional support should just work.

**Reply by PathogenDavid** [reply]
To me that's the difference between "supports" and "works". I feel like I made it pretty clear which C# 9 features *work* with older runtimes but aren't supported by Microsoft.

If you use Roslyn 3.8.0 to compile C# 9.0 code while targeting .NET Framework 4.7.2 and you run into issues, Microsoft will not necessarily help you solve those issues even if you have a support contract with them.
